events/app.py

```diff
@@ -0,0 +1,113 @@
+import os
+import logging
+import datetime
+import pymongo
+# import pprint
+
+from flask import Flask, request, Response
+
+app = Flask(__name__)
+
+# connect to MongoDB Atlas database
+MONGODB_ATLAS_USERNAME = os.environ.get(
+    "MONGODB_ATLAS_USERNAME")
+MONGODB_ATLAS_PASSWORD = os.environ.get(
+    "MONGODB_ATLAS_PASSWORD")
+MONGODB_ATLAS_CLUSTER_ADDRESS = os.environ.get(
+    "MONGODB_ATLAS_CLUSTER_ADDRESS")
+PYMONGO_URI = "mongodb+srv://{}:{}@{}".format(
+    MONGODB_ATLAS_USERNAME,
+    MONGODB_ATLAS_PASSWORD,
+    MONGODB_ATLAS_CLUSTER_ADDRESS)
+# print("Pymongo URI: {}".format(PYMONGO_URI))
+client = pymongo.MongoClient(PYMONGO_URI)
+testDB = client.test
+eventsDB = client.eventsDB
+events_coll = eventsDB.all_events
+
+
+@app.route('/v1/add', methods=['POST'])
+def add_event():
+    # TODO(cmei4444): Get event information from request data
+    info = request.form['info']
+    # Add information into database
+    current_time = datetime.datetime.now()
+    build_event_info(info, current_time)
+    event = Event(info)
+    if not event.is_valid():
+        return Response(
+            status=400,
+            response="Event info was entered incorrectly.",
+        )
+
+    event.add_to_db()
+    return Response(
+        status=201,
+    )
+
+
+def build_event_info(info, time):
+    """Adds created_at time to event info dict"""
+    # MongoDB timestamp precision level, floor to milliseconds
+    time -= datetime.timedelta(0, 0, time.microsecond % 1000)
```

Comment:
You can just assign to `time.microsecond`, FYI.

What does mongodb do if you pass a higher-precision timestamp? Does it error, or just throw away data? If the latter, you might not need to do this.

Reply:
You might try something like
`python
>>> import datetime
>>> datetime.datetime.now().isoformat()
'2019-07-15T13:20:18.085627'
`

Reply:
Do we want to standardize how we represent dates/times across the app? I think ISO format can be hard to parse, so something like `'%Y-%m-%d %H:%M:%S'` would be better, and to convert back and forth we'd use `strftime` and `strptime`

Reply:
That sounds like a good idea, I like your format.